xarray/core/indexing.py

```diff
@@ -345,20 +315,33 @@ def __init__(self, array, key=None):
             Array indexer. If provided, it is assumed to already be in
             canonical expanded form.
         """
-        if key is None:
-            key = (slice(None),) * array.ndim
-        self.array = array
-        self.key = key
+        # We need to avoid doubly wrapping.
+        if isinstance(array, type(self)):
+            self.array = array.array
+            self.key = array.key
+            if key is not None:
+                self.key = self._updated_key(key)
+
+        else:
+            if key is None:
+                key = (slice(None),) * array.ndim
+                key = OuterIndexer(key)
+            self.array = array
+            self.key = key
 
     def _updated_key(self, new_key):
-        new_key = iter(canonicalize_indexer(new_key, self.ndim))
+        # TODO should suport VectorizedIndexer
+        if isinstance(new_key, VectorizedIndexer):
+            raise NotImplementedError('Vectorized indexing for {} is not '
+                                      'implemented.'.format(type(self)))
```

Comment:
This error message needs to give the user guidance, e.g., "Load your data first with .load() or .compute()". Otherwise we will be sure to get users coming to us later for support :)

Reply:
Done